### PR TITLE
Initial Signal integration in sonarr

### PR DESCRIPTION
**Pull request by S0me6uy**
As i did in radarr, here is a PR to support notifications via signal rest API.

Everything is mentionned in https://github.com/Radarr/Radarr/pull/8371

### CODE REVIEW

File: src/NzbDrone.Core/Notifications/Signal/SignalProxy.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            request.SetContent(payload.ToJson());
+            _httpClient.Post(request);
+        }
+
+        public ValidationFailure Test(SignalSettings settings)
+        {
+            try
+            {
+                const string title = "Test Notification";
+                const string body = "This is a test message from Radarr";
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/Signal/SignalProxy.cs:68]: Sonarr

File: src/NzbDrone.Core/Notifications/Signal/SignalPayload.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+namespace NzbDrone.Core.Notifications.Signal
+{
+    public class SignalPayload
+    {
+        public string Message { get; set; }
+
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/Signal/SignalPayload.cs:6]: Nit: no need for new lines between these simple props.

**Review by markus101 — CHANGES_REQUESTED** [review]

File: src/NzbDrone.Core/Notifications/Signal/SignalPayload.cs (hunk 2 above)
Comment (S0me6uy) [reply src/NzbDrone.Core/Notifications/Signal/SignalPayload.cs]: Hello, i changed it in my last commit : is everything ok for you ?

Comment (markus101) [reply src/NzbDrone.Core/Notifications/Signal/SignalPayload.cs]: Yeah, that's great, thanks

**Review by markus101 — APPROVED** [review]
Failing test is related to an issue in `develop`. Thanks for the fixes!

**Pull request merged**